atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueService.java

```diff
@@ -361,4 +370,16 @@ public RowColumnRangeIterator getRowsColumnRange(TableReference tableRef,
                                                                           cellBatchHint,
                                                                           timestamp);
     }
+
+    @Override
+    public void rehydrateLoggingArgSupplier() {
```

Comment:
I guess the KVS is where it gets its data from (since we store the safety in the table metadata). Though agree that after you get the data it doesn't really have much to do with the KVS.

Discussed offline - Actually I think the static class is a pretty good idea. Otherwise things could get pretty messy because say for Background Sweeper to know that the arg-supplier is a thing, you need to pass it to the sweeper... so the arbitrator ends up becoming a super highly linked object.